ts/src/bingx.ts

```diff
@@ -1820,7 +1821,7 @@ export default class bingx extends Exchange {
             if (cost !== undefined) {
                 request['quoteOrderQty'] = this.parseToNumeric (this.costToPrecision (symbol, cost));
             } else {
-                if (market['spot'] && isMarketOrder && (price !== undefined)) {
+                if (isMarketOrder && (price !== undefined) && (side === 'buy')) {
```

Comment:
why are you changing this?

Reply:
we already know that the order is spot because it's inside the spot conditional, and market sell orders require `quantity` to be set

Reply:
@samgermain I think we have a test that contradicts that 
![image](https://github.com/ccxt/ccxt/assets/43336371/3012538a-c9f0-4725-ba2b-1e42fb24bb8b)